keywords/TestServerAndroid.py

```diff
@@ -0,0 +1,244 @@
+            "--es", "username", "none",
+            "--es", "password", "none",
+            "--ei", "listen_port", str(self.port),
+        ])
+        log_info(output)
+        self._wait_until_reachable(port=self.port)
+        self._verify_launched()
+
+        # return "http://{}:{}".format(self.host, self.port)
+
+    def start_device(self, logfile_name):
+        """
+        1. Starts a Test server app with adb logging to provided logfile file object.
+            The adb process will be stored in the self.process property
+        2. Start the Android activity with a launch dictionary
+        2. The method will poll on the endpoint to make sure Test server is available.
+        3. The expected version will be compared with the version reported by http://<host>:<port>
+        4. Return the url of the running Test server app
+        """
+
+        # Clear adb buffer
+        subprocess.check_call(["adb", "-d", "logcat", "-c"])
+
+        # Start redirecting adb output to the logfile
+        self.logfile = open(logfile_name, "w+")
+        self.process = subprocess.Popen(args=["adb", "logcat"], stdout=self.logfile)
+
+        activity_name = "com.couchbase.TestServerApp/com.couchbase.CouchbaseLiteServ.MainActivity"
+        output = subprocess.check_output([
+            "adb", "-d", "shell", "am", "start", "-n", activity_name,
+            "--es", "username", "none",
+            "--es", "password", "none",
+            "--ei", "listen_port", str(self.port),
+        ])
+        log_info(output)
+        self._wait_until_reachable(port=self.port)
+        self._verify_launched()
+
+    def _verify_launched(self):
+        """ Poll on expected http://<host>:<port> until it is reachable
+        Assert that the response contains the expected version information
+        """
+        if self.device_enabled:
+            output = subprocess.check_output(["adb", "-d", "shell", "pidof", "com.couchbase.TestServerApp", "|", "wc", "-l"])
+        else:
+            output = subprocess.check_output(["adb", "-e", "shell", "pidof", "com.couchbase.TestServerApp", "|", "wc", "-l"])
+        log_info("output for running activity {}".format(output))
+        if output is None:
+            raise LiteServError("Err! App did not launched")
+
+    def stop(self):
+        """
+        1. Flush and close the logfile capturing the LiteServ output
+        2. Kill the LiteServ activity and clear the package data
+        3. Kill the adb logcat process
+        """
+
+        log_info("Stopping LiteServ: http://{}:{}".format(self.host, self.port))
+        output = subprocess.check_output(["adb", self.device_option, "shell", "am", "force-stop", "com.couchbase.TestServerApp"])
+        log_info(output)
+
+        # Clear package data
+        output = subprocess.check_output(["adb", self.device_option, "shell", "pm", "clear", "com.couchbase.TestServerApp"])
+        log_info(output)
+
+        # self._verify_not_running()
```

Comment:
Does this need to be uncommented?

Reply:
I am getting request time out when I try to do http GET, so it is not working as expected, I had to comment.